wsi/drm/drmplane.cpp

```diff
@@ -188,7 +195,41 @@ bool DrmPlane::Initialize(uint32_t gpu_fd,
     in_fence_fd_prop_.id = 0;
   }
 
-  return true;
+  // query and store supported modifiers for format, from in_formats
+  // property
+  uint64_t in_formats_prop_value = 0;
+  ret = in_formats_prop_.Initialize(gpu_fd, "IN_FORMATS", plane_props, NULL, &in_formats_prop_value);
+  if (!ret) {
+    ETRACE("Could not get IN_FORMATS property");
+  }
+
+  if (in_formats_prop_value != 0) {
+    drmModePropertyBlobPtr blob = drmModeGetPropertyBlob(gpu_fd, in_formats_prop_value);
+
+    struct drm_format_modifier_blob *m = (struct drm_format_modifier_blob*)(blob->data);
+    struct drm_format_modifier *mod_o =
+      (struct drm_format_modifier *)(((char *)m) + m->modifiers_offset);
```

Comment:
Can u fix the warnings please:

vendor/intel/external/android_ia/hwcomposer/wsi/drm/drmplane.cpp:211:7: warning: cast from 'char *' to 'struct drm_format_modifier *' increases required alignment from 1 to 4 [-Wcast-align]
      (struct drm_format_modifier *)(((char *)m) + m->modifiers_offset);
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
vendor/intel/external/android_ia/hwcomposer/wsi/drm/drmplane.cpp:464:25: warning: comparison of integers of different signs: 'int32_t' (aka 'int') and 'uint32_t' (aka 'unsigned int') [-Wsign-compare]
  for (int32_t i = 0; i < count; i++) {

Reply:
Yes done.

Reply:
vendor/intel/external/android_ia/hwcomposer/wsi/drm/drmplane.cpp:211:7: warning: cast from 'char *' to 'struct drm_format_modifier *' increases required alignment from 1 to 8 [-Wcast-align]
      (struct drm_format_modifier *)(((char *)m) + m->modifiers_offset);

Reply:
Will fix. What environment do you compile? I can not reproduce such warning.